src/Services/DbService/index.js

```diff
@@ -12,18 +12,35 @@ async function getAllStartMarkers(collection) {
     const data = await collection.find({}).toArray();
     let markers = [];
     data.forEach(function (walk) {
-        markers.push({"walkName" : walk.name,
-            "markersObject" : {"lat": walk.markersArray[0].lat, "lng": walk.markersArray[0].lng},
+        markers.push({
+            "walkName" : walk.name,
+            "markersObject" : walk.markersArray[0],
             "id": ObjectId(walk._id)
         });
     })
     return markers;
 }
 
+async function getDogWalkInfo(collection, id) {
+    const o_id = ObjectId(id);
+    return await collection.findOne({'_id': o_id});
+}
+
+async function addAdditionalRouteMarkers(collection, id, newMarkersArray) {
+    const o_id = ObjectId(id);
+    const oldData = await collection.findOne({'_id': o_id});
+    const oldMarkersArray = oldData.markersArray;
+    const updatedMarkersArray = [...oldMarkersArray, ...newMarkersArray];
+    return await collection.updateOne({'_id': o_id}, {$set: {'markersArray': updatedMarkersArray}});
```

Comment:
You might be able to use the push operator here to save a query https://docs.mongodb.com/manual/reference/operator/update/push/